### ISSUE TITLE
White background in MacVim

### ISSUE THREAD

**Issue by Raimondi** [altercation/vim-colors-solarized#9]
I like how this scheme looks, so I want to give it a good test drive, but I'm finding something that hasn't happened before to me.

The status line and folds colors are too close to the background (dark), so it's not easy to distinguish where they are. I often find myself looking for them, with other schemes the status line popped in front of my eyes so it was easy to find the limit between windows, the same with folds.

Here is a  [screenshot](http://cachivaches.chauca.net/files/solarized_vim_stl.png).

**Comment by Raimondi** [comment]
I have changed lines 666 and 667 to this:

```
exe "hi StatusLine"     . s:fg_base02 .s:bg_base0  .s:fmt_none
exe "hi StatusLineNC"   . s:fg_base02 .s:bg_base01 .s:fmt_none
```

after that  I have no problem finding the status line and the folds don't bother me.

Now, I have no idea of what I'm doing, but I thought I should share this anyway :)

**Comment by altercation** [comment]
I've done a fair amount of work on this in the most recent commits. Try the version available now for new styling on active/inactive status bars, horizontal and vertical splits. All should be more visible now.

Let me know if this works for you.

**Comment by Raimondi** [comment]
There's a new issue. Now I get a white background with bg set to light or dark when I launch gvim (MacVim) from Terminal.app which supports 8 (16?) colors. I have no problem with iTerm2 which supports 256 colors, iirc.

Otherwise, I think it looks much better, it's incredible how everything is so easy to see with this color scheme. Thank you very much for your work on this!

**Comment by altercation** [comment]
Is it a white background or the Solarized base03?

**Comment by Raimondi** [comment]
It's white.

**Comment by altercation** [comment]
Can you let me know the values of `echo $TERM` and `tput colors` in Terminal.app?

**Comment by Raimondi** [comment]
```
echo $TERM  => xterm
tput colors      => 8
```

**Comment by altercation** [comment]
Can you try setting the terminal declaration in Terminal.app prefs > settings > advanced to ansi / xterm-color and (using the most recent commit version of solarized.vim) if the results are the same?

**Comment by Raimondi** [comment]
I get the same with both xterm-color and ansi:

http://cachivaches.chauca.net/files/vim-solarized-2011-04-08.png

**Comment by altercation** [comment]
Ah, ok. I thought you meant you were using the macvim vim binary in terminal mode with this problem.

What's the value of `:colorscheme`? Looks to me like it isn't picking it up at all. MacVim should have no issues with displaying the correct values.

**Comment by altercation** [comment]
There is at least one other report of this. Are you all on the current download of MacVim? Confirm build number for me?

**Comment by Raimondi** [comment]
I'm sorry for the confusion, I see the problem when I start MacVim from the terminal (Terminal.app).

`:colorscheme` returns **solarized**, MacVim is 7.3.143.

**Comment by altercation** [comment]
Raimondi, I'm assuming this doesn't happen when you launch from MacVim directly (not via mvim)? Ideally I'd like to chat about this real time. Skype / Jabber? Can you mail me at es@ethanschoonover.com if you're up for it?

**Comment by amaltson** [comment]
I'm having the exact same problem. It's only an issue when you launch it from mvim, not from the dock. More details here: http://stackoverflow.com/questions/5602079/macvim-color-issues-when-launching-from-command-line

**Comment by amaltson** [comment]
Launching it from iTerm2 doesn't have the same issues as launching from Terminal.app.

**Comment by altercation** [comment]
This is caused by `mvim` exporting the terminal environment to MacVim.

I have Solarized check if it's running in Terminal.app and, if so, use a transparent (NONE) background in vim to avoid some weird color issues. Unfortunately I didn't realize that mvim exports it's terminal environment to MacVim as well. I was previously asking vim to check the $TERM_PROGRAM value and if it was terminal.app's string then I assumed I was in it. I've changed this to now check the string and check.

Should MacVim throw the terminal environment away? I don't know. I doubt it. Thus I would recommend best practice in cases where one uses the envrionment variables in vim scripts would be to check for has gui_running as well, if relevant to the check. 

I've added a fix for this as of commit [d93e97f0](https://github.com/altercation/solarized/commit/d93e97f05d7ea7fd8c7d7b79c2b6d522e1ffce1d). Pull or clone the current repo to get the fix.
